core/src/main/java/org/javarosa/core/model/FormDef.java

```diff
@@ -138,10 +144,13 @@
     private final CacheTable<TreeReference, Vector<TreeReference>> cachedCascadingChildren =
             new CacheTable<TreeReference, Vector<TreeReference>>();
 
-
     /**
-     *
+     * Store triggerables that were fired while processing an insert action for
+     * a newly created repeat entry.
      */
+    private Vector<Triggerable> triggerablesProcessedInSetAction =
```

Comment:
This temporary data set and the flag above are fairly coupled. 

Am wondering whether it would be worth it to declare and synchronize on a lock which protects the scoping in which these two values are being used. It would make things much more clear and slightly less error prone. In general it seems good to be protecting the "virtual scope" that is the set of methods that might be active with these members.